src/components/molecules/AssetStatus.tsx

```diff
@@ -0,0 +1,51 @@
+import { DID } from '@oceanprotocol/lib'
+import axios, { CancelTokenSource } from 'axios'
+import React, { ReactElement, useEffect, useState } from 'react'
+import { useOcean } from '../../providers/Ocean'
+import { checkFile } from '../../utils/provider'
+import Status from '../atoms/Status'
+import Tooltip from '../atoms/Tooltip'
+import styles from './AssetStatus.module.css'
+
+export default function AssetStatus({
+  did
+}: {
+  did: DID | boolean
```

Comment:
what's the reasoning behind `DID | boolean`? Makes this prop a bit confusing and I would suggest rather passing the `did` or not based on whatever boolean prop is checked here.

Reply:
The DID is for the compute asset's algorithm list, and the boolean prop is for accepting the `valid` prop from `AssetActions/index`.

Reply:
so let's make that clear in code to get this component more flexible and effective.

1. instead of `did` prop, call it `asset` and have it accept `DDO | DID`. Then when you detect a DDO you skip the `retrieveDDO` call further down. This solves the aquarius double fetching for usages where parent component has the DDO already but still allows refetching in case we do not, like in algo list.
2. still not sure what the boolean is supposed to do, I see this line further down `else if (did === false) setFileValid(did)`. So if your boolean is false you set the file to be valid? If the intention is to signal the component to skip all checks and just set the file to valid, then would use an optional `isFileValid` prop

Reply:
For 2. I am naming the optional prop as `isValid` because we already having: https://github.com/oceanprotocol/market/blob/2ac7f7eb331bb2fa5d02d5199c6643a392d9e92a/src/components/molecules/AssetStatus.tsx#L16